### PR TITLE
QF-2037-Expire-requests-api

### PR DESCRIPTION
**Pull request by karen-hanna**

### CODE REVIEW

File: src/SFA.DAS.RequestApprenticeTraining.Application/Commands/ExpireEmployerRequests/ExpireEmployerRequestsCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            _logger = logger;
+            _applicationSettings = options.Value;
+        }
+
+        public async Task<ExpireEmployerRequestsCommandResponse> Handle(ExpireEmployerRequestsCommand request, CancellationToken cancellationToken)
+        {
+            await _employerRequestEntityContext.ExpireEmployerRequests(_applicationSettings.ExpiryAfterMonths);
+            await _employerRequestEntityContext.SaveChangesAsync();
+
+            return new ExpireEmployerRequestsCommandResponse();
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.RequestApprenticeTraining.Application/Commands/ExpireEmployerRequests/ExpireEmployerRequestsCommandHandler.cs]: There is no information in the response so this is a command which does not have a response

File: src/SFA.DAS.RequestApprenticeTraining.Domain/Entities/EmployerRequest.cs [hunk 2]
```diff
@@ -22,16 +22,18 @@ public class EmployerRequest
...
         public List<ProviderResponseEmployerRequest> ProviderResponseEmployerRequests { get; set; }
-      
+
         public List<Region> GetRegions()
         {
             return EmployerRequestRegions.Select(err => err.Region).ToList();
         }
         public Standard Standard { get; set; }
-        
+
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.RequestApprenticeTraining.Domain/Entities/EmployerRequest.cs]: Unnecessary white space

File: src/SFA.DAS.RequestApprenticeTraining.Domain/Interfaces/IEmployerRequestEntityContext.cs [hunk 3]
```diff
@@ -149,5 +149,25 @@ public async Task<List<SelectEmployerRequest>> GetForProviderResponse(Guid provi
...
+            var result = await Entities
+                .Where(er => er.RequestStatus == Models.Enums.RequestStatus.Active
+                    && er.RequestedAt < expiryRequestsRequestedBeforeDate)
+                .ToListAsync();
+
+            result.ForEach(er => 
+            {
+                er.RequestStatus = Models.Enums.RequestStatus.Expired;
+                er.ExpiredAt = dateTimeNow;
+
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.RequestApprenticeTraining.Domain/Interfaces/IEmployerRequestEntityContext.cs]: White space

File: src/SFA.DAS.RequestApprenticeTraining.Domain/Interfaces/IEmployerRequestEntityContext.cs [hunk 4]
```diff
@@ -149,5 +149,25 @@ public async Task<List<SelectEmployerRequest>> GetForProviderResponse(Guid provi
...
+            result.ForEach(er => 
+            {
+                er.RequestStatus = Models.Enums.RequestStatus.Expired;
+                er.ExpiredAt = dateTimeNow;
+                
+            });
+                
+        }
+
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.RequestApprenticeTraining.Domain/Interfaces/IEmployerRequestEntityContext.cs]: White space

File: src/SFA.DAS.RequestApprenticeTraining.Domain/Configuration/ApplicationSettings.cs [hunk 5]
```diff
@@ -9,6 +9,10 @@ public class ApplicationSettings
         public string DbConnectionString { get; set; }
         public string NServiceBusConnectionString { get; set; }
         public string NServiceBusLicense { get; set; }
+        public int ExpiryAfterMonths { get; set; }
+        public int EmployerRemovedAfterExpiryMonths { get; set; }
+        public int ProviderRemovedAfterRequestedMonths { get; set; }
+
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.RequestApprenticeTraining.Domain/Configuration/ApplicationSettings.cs]: White space

**Review by ChrisJWoodcock — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.RequestApprenticeTraining.Application/Commands/ExpireEmployerRequests/ExpireEmployerRequestsCommandHandler.cs (hunk 1 above)
Comment (karen-hanna) [reply src/SFA.DAS.RequestApprenticeTraining.Application/Commands/ExpireEmployerRequests/ExpireEmployerRequestsCommandHandler.cs]: removed empty response

File: src/SFA.DAS.RequestApprenticeTraining.Domain/Entities/EmployerRequest.cs (hunk 2 above)
Comment (karen-hanna) [reply src/SFA.DAS.RequestApprenticeTraining.Domain/Entities/EmployerRequest.cs]: removed

File: src/SFA.DAS.RequestApprenticeTraining.Domain/Interfaces/IEmployerRequestEntityContext.cs (hunk 3 above)
Comment (karen-hanna) [reply src/SFA.DAS.RequestApprenticeTraining.Domain/Interfaces/IEmployerRequestEntityContext.cs]: removed

File: src/SFA.DAS.RequestApprenticeTraining.Domain/Interfaces/IEmployerRequestEntityContext.cs (hunk 4 above)
Comment (karen-hanna) [reply src/SFA.DAS.RequestApprenticeTraining.Domain/Interfaces/IEmployerRequestEntityContext.cs]: removed

File: src/SFA.DAS.RequestApprenticeTraining.Domain/Configuration/ApplicationSettings.cs (hunk 5 above)
Comment (karen-hanna) [reply src/SFA.DAS.RequestApprenticeTraining.Domain/Configuration/ApplicationSettings.cs]: removed

**Review by ChrisJWoodcock — DISMISSED** [review]

**Review by ChrisJWoodcock — APPROVED** [review]

**Pull request merged**